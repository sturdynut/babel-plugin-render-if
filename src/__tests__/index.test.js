import React from "react";
import * as babel from "@babel/core";
import plugin from "../";

describe("RenderIf", () => {
  const options = { plugins: [plugin] };

  const transform = (renderIf) => babel.transform(renderIf, options);

  describe("with text child", () => {
    const renderIf = `<RenderIf isTrue={isTextChild}>THE CHILDREN</RenderIf>`;

    it("transforms as expected", () => {
      const { code } = transform(renderIf);
      expect(code).toMatchSnapshot();
    });
  });

  describe("with single child", () => {
    const renderIf = `<RenderIf isTrue={isSingleChild}>
        <span>THE CHILDREN</span>
      </RenderIf>`;

    it("transforms as expected", () => {
      const { code } = transform(renderIf);
      expect(code).toMatchSnapshot();
    });
  });

  describe("with nested children", () => {
    const renderIf = `<RenderIf isTrue={isNestChildren}>
        <div><span>THE CHILDREN</span></div>
      </RenderIf>`;

    it("transforms as expected", () => {
      const { code } = transform(renderIf);
      expect(code).toMatchSnapshot();
    });
  });

  describe("with array children", () => {
    const renderIf = `<RenderIf isTrue={isNestChildren}>
        {[
          <span>CHILD 1</span>,
          <span>CHILD 2</span>,
        ]}
      </RenderIf>`;

    it("transforms as expected", () => {
      const { code } = transform(renderIf);
      expect(code).toMatchSnapshot();
    });
  });

  describe("with js expression as child", () => {
    const renderIf = `<RenderIf isTrue={isNestChildren}>{\`foo\`}</RenderIf>`;

    it("transforms as expected", () => {
      const { code } = transform(renderIf);
      expect(code).toMatchSnapshot();
    });
  });

  describe("with no children", () => {
    const renderIf = `<RenderIf isTrue={isNestChildren}></RenderIf>`;

    it("throws error", () => {
      expect(() => transform(renderIf)).toThrowError(
        "Missing children for RenderIf"
      );
    });
  });

  describe("with no condition", () => {
    const renderIf = `<RenderIf>BLOW UP!</RenderIf>`;

    it("throws error", () => {
      expect(() => transform(renderIf)).toThrowError(
        "Missing condition for RenderIf"
      );
    });
  });
});
