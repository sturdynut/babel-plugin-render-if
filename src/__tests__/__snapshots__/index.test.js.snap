// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RenderIf with array children transforms as expected 1`] = `"isNestChildren ? [<span>CHILD 1</span>, <span>CHILD 2</span>] : null;"`;

exports[`RenderIf with js expression as child transforms as expected 1`] = `"isNestChildren ? \`foo\` : null;"`;

exports[`RenderIf with nested children transforms as expected 1`] = `"isNestChildren ? <div><span>THE CHILDREN</span></div> : null;"`;

exports[`RenderIf with single child transforms as expected 1`] = `"isSingleChild ? <span>THE CHILDREN</span> : null;"`;

exports[`RenderIf with text child transforms as expected 1`] = `"isTextChild ? \\"THE CHILDREN\\" : null;"`;
